okhttp/src/main/java/io/grpc/okhttp/OkHttpProtocolNegotiator.java

```diff
@@ -235,7 +236,9 @@ protected void configureTlsExtensions(
       SSLParameters sslParams = sslSocket.getSSLParameters();
       try {
         // Enable SNI and session tickets.
-        if (hostname != null) {
+        // Skip if hostname is not a real internet domain name, which is usually used with
+        // test certificates.
```

Comment:
Hmmm, don't like the idea of adding "special" code/workaround in main code to make tests pass. Why not modify test certs so they don't need special handling in mainline code?

Reply:
Details in b/154375837. According to @ejona86, changing those test certs seems to be hard.

Reply:
FWIW, SunJSSE ignores SNI if it is invalid (at least in some ways): https://github.com/grpc/grpc-java/issues/4912#issuecomment-429138168 .

While I could potentially agree using IDN is a good idea, it is a bit more subtle of a change. How would you feel about calling GrpcUtil.checkAuthority() (with try/catch) here instead? Then we could say that checkAuthority is [normally performed in the builder](https://github.com/grpc/grpc-java/blob/4d3435056b81aab764433d022b91f3b5a7279f98/core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java#L505) so it should virtually always succeed. We then reference the bug id, and say we are working around cases where checkAuthority is disabled and we want to avoid them causing trouble here. We can also mention that underscore is the troublesome character.

Doing it that way makes it much more obvious this changes no behavior change for most users and makes it more obvious if/when it is no longer needed (e.g., because the broken certs have been replaced).